File: src/main/java/com/vc/easy/L458.scala
package com.vc.easy

object L458 {
  def poorPigs(buckets: Int, minutesToDie: Int, minutesToTest: Int): Int = {
    val time = minutesToTest / minutesToDie
    val attempt = time + 1
    var pigs = 0
    while(Math.pow(attempt, pigs) < buckets) pigs += 1
    pigs
  }
}

File: src/main/java/com/vc/hard/L458.java
package com.vc.hard;

class L458 {
    public int poorPigs(int buckets, int minutesToDie, int minutesToTest) {
        int times = minutesToTest / minutesToDie + 1;
        int pig = 0;
        while(Math.pow(times, pig) < buckets) {
            pig++;
        }
        return pig;
    }
}
